apps/chat/src/store/conversations/conversations.epics.ts

```diff
@@ -1113,76 +1114,109 @@ const sendMessageEpic: AppEpic = (action$, state$) =>
+      ),
     })),
-    map(({ payload, modelsMap, conversations, selectedConversationIds }) => {
-      const messageModel: Message[EntityType.Model] = {
-        id: payload.conversation.model.id,
-      };
-      const messageSettings: Message['settings'] = {
-        prompt: payload.conversation.prompt,
-        temperature: payload.conversation.temperature,
-        selectedAddons: payload.conversation.selectedAddons,
-        assistantModelId: payload.conversation.assistantModelId,
-      };
+    map(
+      ({
+        payload,
+        modelsMap,
+        conversations,
+        selectedConversationIds,
+        overlaySystemPrompt,
+      }) => {
+        const messageModel: Message[EntityType.Model] = {
+          id: payload.conversation.model.id,
+        };
+        const messageSettings: Message['settings'] = {
+          prompt: payload.conversation.prompt,
+          temperature: payload.conversation.temperature,
+          selectedAddons: payload.conversation.selectedAddons,
+          assistantModelId: payload.conversation.assistantModelId,
+        };
 
-      const assistantMessage: Message = {
-        content: '',
-        model: messageModel,
-        settings: messageSettings,
-        role: Role.Assistant,
-      };
+        let currentMessages =
+          payload.deleteCount > 0
+            ? payload.conversation.messages.slice(
+                0,
+                payload.deleteCount * -1 || undefined,
+              )
+            : payload.conversation.messages;
 
-      const userMessage: Message = {
-        ...payload.message,
-        model: messageModel,
-        settings: messageSettings,
-      };
+        const assistantMessage: Message = {
+          content: '',
+          model: messageModel,
+          settings: messageSettings,
+          role: Role.Assistant,
+        };
 
-      const updatedMessages: Message[] = (
-        payload.deleteCount > 0
-          ? payload.conversation.messages.slice(
-              0,
-              payload.deleteCount * -1 || undefined,
-            )
-          : payload.conversation.messages
-      ).concat(userMessage, assistantMessage);
+        const userMessage: Message = {
+          ...payload.message,
+          model: messageModel,
+          settings: messageSettings,
+        };
 
-      const newConversationName = getNextDefaultName(
-        getNewConversationName(
-          payload.conversation,
-          payload.message,
-          updatedMessages,
-        ),
-        conversations.filter(
-          (conv) =>
-            conv.folderId === payload.conversation.folderId &&
-            conv.id !== payload.conversation.id,
-        ),
-        Math.max(selectedConversationIds.indexOf(payload.conversation.id), 0),
-        true,
-      );
+        /*
+          Overlay needs to share host application state information
+          We storing state information in systemPrompt (message with role: Role.System)
+        */
+        if (overlaySystemPrompt) {
```

Comment:
please add addditional check is it overlay or not just to be sure that it will not called in normal mode

Moreover please extract this to some method outside of this epic file

Reply:
Fixed.